src/simulation/dynamics/constraintEffector/constraintDynamicEffector.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ ISC License
+
+ Copyright (c) 2022, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include "constraintDynamicEffector.h"
+#include "architecture/utilities/avsEigenSupport.h"
+
+/*! This is the constructor, setting variables to default values */
+ConstraintDynamicEffector::ConstraintDynamicEffector()
+{ 
+    // Initialize counters and flags
+    this->scInitCounter = 0;
+    this->scID = 1;
+
+    // Initialize constraint dimensions
+    this->r_P1B1_B1.setZero();
+    this->r_P2B2_B2.setZero();
+    this->r_P2P1_B1Init.setZero();
+
+    // Initialize gains
+    this->alpha = 0.0;
+    this->beta = 0.0;
+    this->k_d = 0.0;
+    this->c_d = 0.0;
+    this->k_a = 0.0;
+    this->c_a = 0.0;
+
+    // Initialize the force and torque quantities
+    this->Fc_N.setZero();
+    this->L_B2.setZero();
+}
+
+/*! This is the destructor, nothing to report here */
+ConstraintDynamicEffector::~ConstraintDynamicEffector()
+{
+    
+}
+
+/*! This method is used to reset the module.
+ @return void
+ */
+void ConstraintDynamicEffector::Reset(uint64_t CurrentSimNanos)
+{
+    this->k_d = pow(this->alpha,2);
```

Comment:
I don't like having the gains set inside the Reset() method.  Reset can be called at various time and this can lead to unexpected behavior.  As we are introducing setter and getter methods to this module, let's set the gains inside a setter function to specify alpha and beta.  This way Reset() can be called without changing the gains applied.